ibis/backends/dask/execution/selection.py

```diff
@@ -137,6 +141,9 @@ def execute_selection_dataframe(
 
     # Build up the individual dask structures from column expressions
     if selections:
+        # Create a unique row identifier and set it as the index. This is used
+        # in dd.concat to merge the pieces back together.
+        data = add_partitioned_sorted_column(data)
         data_pieces = []
         for selection in selections:
             dask_object = compute_projection(
```

Comment:
I think I missed some points here in previous reviews:

1. Could `compute_project` changes the index? i.e, does the index on `dask_object` is always the same?
2. Where is the original columns in `data`? it seems it is not part of `data_pieces` so not clear to me how did we keep the original columns

Reply:
> 1. Could compute_project changes the index? i.e, does the index on dask_object is always the same?

Currently there is no operation that changes the index. Grouped window will, but we will take care of alignment in the udf. 

> 2. Where is the original columns in data? it seems it is not part of data_pieces so not clear to me how did we keep the original columns

I don't really understand the question here. Can you elaborate?

Reply:
> Currently there is no operation that changes the index. Grouped window will, but we will take care of alignment in the udf.

Got it. If that is an invariant of the code here (compute_projection does not change index), I think it's probably a good idea to clearly document that in the code here. (Better if we can assert in compute_projection that this invariant is True, can be follow up).

Reply:
> I don't really understand the question here. Can you elaborate?

nvm. I figured it out...

I think what's confusing is that selection[0] (the parent table) is not a Projection but is being passed to the function called `compute_projection`, which confused me.  Might be good to add some comments here for future readers.

Reply:
`selection` is an expression, that gets passed, along with `data` (the parent table), to `compute_projection` which is a dispatcher that is registered against function performing the action: https://github.com/ibis-project/ibis/blob/master/ibis/backends/dask/execution/selection.py#L31

Reply:
I see. Consider my point (2) addressed